src/configure/configure.ts

```diff
@@ -444,16 +449,49 @@ class PipelineConfigurer {
         this.inputs.azureSession = getSubscriptionSession(this.inputs.targetResource.subscriptionId);
 
         // show available resources and get the chosen one
-        this.appServiceClient = new AppServiceClient(this.inputs.azureSession.credentials, this.inputs.targetResource.subscriptionId);
+        this.appServiceClient = new AppServiceClient(this.inputs.azureSession.credentials, this.inputs.azureSession.tenantId, this.inputs.azureSession.environment.portalUrl, this.inputs.targetResource.subscriptionId);
         let selectedResource: QuickPickItemWithData = await this.controlProvider.showQuickPick(
             constants.SelectWebApp,
             this.appServiceClient.GetAppServices(WebAppKind.WindowsApp)
                 .then((webApps) => webApps.map(x => { return { label: x.name, data: x }; })),
             { placeHolder: Messages.selectWebApp },
             TelemetryKeys.WebAppListCount);
+
         this.inputs.targetResource.resource = selectedResource.data;
     }
 
+    private async updateScmType(queuedPipeline: Build): Promise<void> {
+        // update SCM type
+        this.appServiceClient.updateScmType(this.inputs.targetResource.resource.id);
+
+        let buildDefinitionUrl = this.azureDevOpsClient.getOldFormatBuildDefinitionUrl(this.inputs.organizationName, this.inputs.project.id, queuedPipeline.definition.id);
+        let buildUrl = this.azureDevOpsClient.getOldFormatBuildUrl(this.inputs.organizationName, this.inputs.project.id, queuedPipeline.id);
+
+        // update metadata of app service to store information about the pipeline deploying to web app.
+        new Promise<void>(async (resolve) => {
+            let metadata = await this.appServiceClient.getAppServiceMetadata(this.inputs.targetResource.resource.id);
+            let organizationId = await this.azureDevOpsClient.getOrganizationIdFromName(this.inputs.organizationName);
+            metadata["properties"] = {
```

Comment:
you should override individual  properties  and not override the properties object itself. There could be some useful information being used by other service